### ISSUE TITLE
Is it possible to add like operator support in Mongo Visitor?

### ISSUE THREAD

**Issue by sashajo** [RutledgePaulV/q-builders#18]
It would be nice if there is a way to support like or regex search via rsql.
Either field==f* or field=like=f* or field=regex=^f.* etc.

This would also help when searching for empty string for example if the field ="" , there is no way to express query to return object with field with empty string that I am aware of.

Many thanks

**Comment by RutledgePaulV** [comment]
Hi Sashajo. Yes this is possible. I've received this request a couple times now so I think I'll just go ahead and implement it. The reason I held off originally is due to the complexity around supporting regex searches against multiple backends. (elasticsearch, mongo, and java all handle them somewhat differently). I'll likely implement it so it just passes through whatever value is specified - meaning that if you express it in Mongo form, you can't expect it to work in all cases for the Java predicates, etc.

**Comment by RutledgePaulV** [comment]
I've added regex support for string fields. It will be released under 1.4

```
Condition<User> condition = name().pattern(".*ajo");
Criteria criteria = condition.query(new MongoVisitor());
criteria.toString(); // "{ \"name\" : { \"$regex\" : \".*ajo\"}}";
```
